Bigtable/src/DataClient.php

```diff
@@ -199,67 +210,91 @@ public function upsert(array $rows, array $options = [])
+     *           (`endKeyOpen` or `endKeyClosed`).
+     *     @type int $rowsLimit The number of rows to scan.
+     * }
+     * @return ChunkFormatter
      */
     public function readRows(array $options = [])
     {
-        $rowSet = null;
-        $rowKeys = $this->pluck('rowKeys', $options, false);
-        $ranges = $this->pluck('rowRanges', $options, false);
-        if ($rowKeys !== null) {
-            $rowSet = new RowSet();
-            if (is_array($rowKeys)) {
-                $rowSet->setRowKeys($rowKeys);
-            } else {
-                $rowSet->setRowKeys([$rowKeys]);
-            }
-        }
-        if ($ranges !== null) {
-            if ($rowSet === null) {
-                $rowSet = new RowSet();
-            }
-            $rowRanges = [];
-            foreach ($ranges as $range) {
-                $rowRange = new RowRange();
-                if (isset($range['startKeyOpen'])) {
-                    $rowRange->setStartKeyOpen($range['startKeyOpen']);
-                } elseif (isset($range['startKeyClosed'])) {
-                    $rowRange->setStartKeyClosed($range['startKeyClosed']);
-                }
-                if (isset($range['endKeyOpen'])) {
-                    $rowRange->setEndKeyOpen($range['endKeyOpen']);
-                } elseif (isset($range['endKeyClosed'])) {
-                    $rowRange->setEndKeyClosed($range['endKeyClosed']);
-                }
-                $rowRanges[] = $rowRange;
-            }
-            $rowSet->setRowRanges($rowRanges);
-        }
-        $args = $options + $this->options;
-        if ($rowSet !== null) {
-            $args['rows'] = $rowSet;
+        $rowKeys = $this->pluck('rowKeys', $options, false) ?: [];
+        $ranges = $this->pluck('rowRanges', $options, false) ?: [];
+
+        array_walk($ranges, function (&$range) {
+            $range = $this->serializer->decodeMessage(
+                new RowRange(),
+                $range
+            );
+        });
+
+        if ($ranges || $rowKeys) {
+            $options['rows'] = $this->serializer->decodeMessage(
+                new RowSet,
+                [
+                    'rowKeys' => $rowKeys,
+                    'rowRanges' => $ranges
+                ]
+            );
         }
+
         $serverStream = $this->bigtableClient->readRows(
             $this->tableName,
-            $args
+            $options + $this->options
         );
-        $chunkFormatter = new ChunkFormatter($serverStream);
-        return $chunkFormatter;
+        return new ChunkFormatter($serverStream);
+    }
+
+    /**
+     * Read a single row from the table.
+     *
+     * Example:
+     * ```
+     * $row = $dataClient->readRow('jefferson');
+     *
+     * print_r($row);
+     * ```
+     *
+     * @param string $rowKey The row key to read.
+     * @param array $options [optional] Configuration options.
+     * @return array|null
+     */
+    public function readRow($rowKey, array $options = [])
```

Comment:
Having a separate method for reading a single row seemed a little easier to reason about in my head than accepting either a string or array for `rowKeys` in `readRows`. WDYT?

Reply:
Thanks for putting this. We had a plan to add this after readRows.